agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/choose-house-card-game-state/ChooseHouseCardGameState.ts

```diff
@@ -139,9 +168,23 @@ export default class ChooseHouseCardGameState extends GameState<CombatGameState>
         }
     }
 
+    canRefuseSupport(house: House): boolean {
+        // Support can only be refused if house is supported and if it has not played a house card yet
+        return this.combatGameState.supporters.values.includes(house) && !this.houseCards.has(house);
+    }
+
+    refuseSupport(): void {
+        this.entireGame.sendMessageToServer({
+            type: "refuse-support"
+        });
+    }
+
     private tryAutomaticallyChooseLastHouseCard(house: House): void {
         const choosableCards = this.getChoosableCards(house);
-        if (choosableCards.length == 1) {
+
+        // We can only fast-track the last house card if house received no support to allow refusing it here
+        if (choosableCards.length == 1 &&
+            !this.combatGameState.supporters.values.includes(house)) {
```

Comment:
Its also possible to fast track if a player only supports themselves, not only if they get not support at all

Reply:
Well, I guess we have to keep it that way because you automatically support yourself now and you might also want to refuse this automatically granted support before playing e.g. Roose Bolton as your last card to lose the battle and get Roose back instead of discarding him on battle win.

But you think the rest of the work flow is ok? Were you able to test it already?